### DISCUSSION TITLE
Task vs ValueTask - The what's, when's and how's

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#11] [General]
# Task vs ValueTask - The what's, when's and how's

Since .NET Core 2.0 the Type ValueTask. It seems that there is a lot of overlap between Task and ValueTask.
So let's have a deeper look into ValueTask. Where should we use it and how should we use it properly? And also: where we should not use is.

https://steven-giesel.com/blogPost/28ed9a4d-c390-46f1-ad83-8fda4747ed1f

**Comment by Yoannnn** [comment]
Hi,
I think there is a missing "return" in front of _weatherCache[city]; in the paragraph "Wrapping TResult and Task<TResult>".
Thanks for your article !
Regards,
Yoann.

**Reply by linkdotnet** [reply]
Oh boy thanks :D yes I missed that.
